[Onboard] add new sample of Office Add-in (#1372)
OfficeDev/teams-toolkit-samples: hello-world-office-addin/README.md

---
page_type: sample
languages:
- typescript
products:
- office
name: Hello World Office add-in
urlFragment: officedev-teamsfx-samples-hello-world-office-add-in
description: A hello world project that contains Office add-in capability.
extensions:
  createdDate: "2024-12-06"
---
# Getting Started with Hello World Office Add-in Sample

Office add-ins are integrations built by third parties into Office by using our web-based platform.

> 

Reviewer: ```suggestion
```

> Now you have the ability to create a single unit of distribution for all your Office applications by using the same manifest format and schema, based on the current JSON-formatted MetaOS manifest.

Reviewer: With the unified app manifest, you can create a single unit of distribution for all your Microsoft 365 extensions by using the same manifest format and schema.

Currently unified manifest of Office add-in is in public preview, so it does not support deploy and publish.

## This sample illustrates

> - How to create an Office add-ins with a unified manifest that works on Word, Excel, Powerpoint, and Outlook. 

Reviewer: ```suggestion
- How to create an Office add-in with a unified manifest that works on Word, Excel, Powerpoint, and Outlook. 
```

## Minimal path to awesome

### Debug Office add-in

- Please note that the same M365 account should be used both in Teams Toolkit and Office Applications (Word, Excel, Powerpoint, Outlook).
- From Visual Studio Code only: use the `Run and Debug Activity Panel` in Visual Studio Code, select `Debug in Excel Desktop (Edge Chromium)`, and click the `Run and Debug` green arrow button. Please run VSCode as administrator if localhost loopback for Microsoft Edge Webview hasn't been enabled. Once enbaled, administrator priviledge is no longer required.

   ![Visual Studio Code debug configuration for Office](./assets/launch.png)

>    Once the Office app is open, you can select the ribbon command to show the add-in task pane. The snapshot shows Excel as an example, and the other Office applications share the same behavior. 

Reviewer: ```suggestion
   Once the Office app is open, you can select the ribbon command to show the add-in task pane. 
```

>    ![Excel add-in show taskpane](./assets/excel-ribbon.png)

Reviewer: Same comment here

>    The taskpane opens and you can check the sideloaded add-in.

Reviewer: ```suggestion
- The taskpane opens and you can check the sideloaded add-in.
```

>    ![Excel add-in task pane opened](./assets/thumbnail.png)

Reviewer: Same here

> 

Reviewer: ```suggestion
```

### Edit the manifest

> You can find the app manifest in `./appPackage` folder. The folder contains one manifest file:

Reviewer: ```suggestion
You can find the app manifest in `./appPackage` folder. The folder contains the manifest file `manifest.json` for Office add-in running locally.
```

> - `manifest.json`: Manifest file for Office add-in running locally.

Reviewer: ```suggestion

```

You may add any extra properties or permissions you require to this file. See the [schema reference](https://docs.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema) for more information.

> 

Reviewer: ```suggestion
```

- Edge installed for debugging Office add-in.
- Office for Windows: Beta Channel. Follow [this link](https://github.com/OfficeDev/TeamsFx/wiki/How-to-switch-Outlook-client-update-channel-and-verify-Outlook-client-build-version) for information about how to update channels and check your Office client build version.
- An M365 account. If you do not have M365 account, apply one from [M365 developer program](https://developer.microsoft.com/en-us/microsoft-365/dev-program)
- [Teams Toolkit Visual Studio Code Extension](https://aka.ms/teams-toolkit) version 5.0.0 and higher or [Teams Toolkit CLI](https://aka.ms/teams-toolkit-cli)

## Minimal path to awesome

### Debug Office add-in

- Please note that the same M365 account should be used both in Teams Toolkit and Office Applications (Word, Excel, Powerpoint, Outlook).
- From Visual Studio Code only: use the `Run and Debug Activity Panel` in Visual Studio Code, select `Debug in Excel Desktop (Edge Chromium)`, and click the `Run and Debug` green arrow button. Please run VSCode as administrator if localhost loopback for Microsoft Edge Webview hasn't been enabled. Once enbaled, administrator priviledge is no longer required.

>    ![Visual Studio Code debug configuration for Office](./assets/launch.png)

Reviewer: ```suggestion

```
Seems the image does not exist

   Once the Excel app is open, you can select the option to show a task pane.
   ![Excel add-in show taskpane](./assets/excel-ribbon.png)

   The taskpane should look as shown in the following image.
   ![Excel add-in task pane opened](./assets/thumbnail.png)

   Other Office Applications share the same behavior.

### Edit the manifest

You can find the app manifest in `./appPackage` folder. The folder contains one manifest file:

- `manifest.json`: Manifest file for Office add-in running locally.

You may add any extra properties or permissions you require to this file. See the [schema reference](https://docs.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema) for more information.

> ### Deploy and publish

Reviewer: ```suggestion

```

> Currently unified manifest of Office add-in is in public preview, so it does not support deploy and publish.

Reviewer: ```suggestion

```

Reviewer: put this at front
